Sources/SourceControl/RepositoryManager.swift

```diff
@@ -230,10 +233,20 @@ public class RepositoryManager {
     /// - Throws:
     /// - Returns: Details about the performed fetch.
     @discardableResult
-    func fetchAndPopulateCache(handle: RepositoryHandle, repositoryPath: AbsolutePath) throws -> FetchDetails {
+    func fetchAndPopulateCache(handle: RepositoryHandle, repositoryPath: AbsolutePath, on queue: DispatchQueue) throws -> FetchDetails {
```

Comment:
iiuc the queue is for the delegate, not the action? if so, should it be `delegateQueue: DispatchQueue`

Reply:
done 7983807